### ISSUE TITLE
Rasa X imports responses as actions in domain data from Github sync

### ISSUE THREAD

**Issue by ig-shaun** [RasaHQ/rasa#5979]
**Rasa version**: 1.10.2

**Rasa SDK version** (if used & relevant):

**Rasa X version** (if used & relevant): 

**Python version**:

**Operating system** (windows, osx, ...):

**Issue**:
Domain in X syncs incorrectly with the Git master domain file. 
It pulls responses into actions. 
So I end up with -utter_ statements sitting in the incorrect place. 
This prevents the model from training. (On running 'train' the generic "nothing has changed … so we skipped training a new one" message appears).
Without making any changes, when I sync the domain file that X generates back into a new git branch, this also has the responses incorrectly listed under actions. If I then delete the incorrectly located -utter_ responses within the X domain and run train, this works. But only temporarily, as the error recurs with the next X sync back to Master. Responses again get placed under actions - even though the Git Master file is correct. 

**Error (including full traceback)**:
```

```

**Command or request that led to error**:
```

```

**Content of configuration file (config.yml)** (if relevant):
```yml

```

**Content of domain file (domain.yml)** (if relevant):
```yml
https://github.com/ixoworld/assistant/blob/master/domain.yml
```

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @Ghostvv will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by TyDunn** [comment]
@ig-shaun What version of Rasa X are you using?

**Comment by ig-shaun** [comment]
I installed the latest version using the one-line install script a couple of days back

**Comment by TyDunn** [comment]
I was able to reproduce your issue. That is a bug in Rasa X, which we need to squash
![image](https://user-images.githubusercontent.com/13314504/84258898-fa4a6b00-ab17-11ea-8019-8593cc2fd127.png)

**Comment by chkoss** [comment]
I understand that you
- had a domain and a model that was trained on that domain
- started Rasa X, which changed the domain in the way you described (adding the responses to `actions`)
- the altered domain was synced back to your domain file in master
- you run `rasa train` again and get the message "nothing has changed … so we skipped training a new one"

Is that correct or did you change anything else manually in the domain, stories or nlu examples between training the last model and running `rasa train` again?

If you didn't change anything else, then the message you saw for `rasa train` is the correct behaviour:
The `utter_` statements that are listed under responses are a specific type of action. They don't have to be listed in the `actions` section because rasa knows automatically that all responses are actions, but they can be listed there. So adding them to the `actions` section in the domain isn't wrong. A domain with or without the responses listed under `actions` is still "the same" domain, so when running `rasa train` with this being the only change in the training data, no training needs to be done and you should get that message.

I still agree that it's not nice if the domain file gets unexpectedly modified, and we should fix that.

**Comment by ig-shaun** [comment]
The only difference from what you described is that I did not merge the branch with RasaX-generated changes (addition of responses under actions in domain) back to Master.

I believe that your expected behaviour is not functioning how you describe, as I am able only able to train when I make a small change in for instance the NLU and delete the inserted responses before running Train. 

On the next sync with master the responses get added again and if I make the same NLU change it won’t train (gives the no changes message). 

Perhaps if I could stop the Auto sync then I could go ahead to make changes, then train and save as a new branch and merge to master before doing another sync. 

I am available to run through a remote test if necessary. Really need to get this issue resolved soon or we will have to abandon plans to include the assistant in our software release.

**Comment by chkoss** [comment]
Opened an issue for Rasa Open Source on preventing the addition of responses into `actions`: https://github.com/RasaHQ/rasa/issues/6033

**Comment by InnoOmnia** [comment]
I have the same issue, I started from scratch, connected to github, and here is the result
session_config:
  session_expiration_time: 60
  carry_over_slots_to_new_session: true
intents:
  - greet
  - affirm
  - deny
  - kurssi
responses:
  utter_goodbye:
    - text: Bye
  utter_greet:
    - text: Hey! How are you?
  utter_1_bookkeep_bascis:
    - text: Hallitsen kirjanpidon perusteet.
  utter_2_monthly_updates:
    - text: Osaan tehdä normaalit kuukausittaiset kirjanpidon kirjaukset.
  utter_3_bookkeeping_special:
    - text: Osaan tehdä kirjanpitoa huomioiden erityistilanteet.
  utter_tervetuloa:
    - text: Tämä on taloushallinnon testi. Vastaa kysymyksiin kyllä tai ei
actions:
  - utter_goodbye
  - utter_greet
  - utter_1_bookkeep_bascis
  - utter_2_monthly_updates
  - utter_3_bookkeeping_special
  - utter_tervetuloa

**Comment by InnoOmnia** [comment]
I  have same issue, I started from scratch, created github link, and the looked config file
intents:
  - greet
  - affirm
  - deny
  - kurssi

responses:
  utter_goodbye:
    - text: Bye
  utter_greet:
    - text: Hey! How are you?
  utter_1_bookkeep_bascis:
    - text: Hallitsen kirjanpidon perusteet.
  utter_2_monthly_updates:
    - text: Osaan tehdä normaalit kuukausittaiset kirjanpidon kirjaukset.
  utter_3_bookkeeping_special:
    - text: Osaan tehdä kirjanpitoa huomioiden erityistilanteet.
  utter_tervetuloa:
    - text: Tämä on taloushallinnon testi. Vastaa kysymyksiin kyllä tai ei



actions:
  - utter_goodbye
  - utter_greet
  - utter_1_bookkeep_bascis
  - utter_2_monthly_updates
  - utter_3_bookkeeping_special
  - utter_tervetuloa

**Comment by ricwo** [comment]
fixed in rasa X 0.31.4 https://rasa.com/docs/rasa-x/changelog/rasa-x-changelog/#id1
